pkg/controller/pxcrestore/restorer.go

```diff
@@ -0,0 +1,348 @@
+}
+
+func (s *s3) PITRJob() (*batchv1.Job, error) {
+	return backup.RestoreJob(s.cr, s.bcp, s.cluster, strings.TrimPrefix(s.bcp.Status.Destination, api.AwsBlobStoragePrefix), true)
+}
+
+func (s *s3) ValidateJob(ctx context.Context, job *batchv1.Job) error {
+	if s.bcp.Status.S3.CredentialsSecret == "" {
+		// Skip validation if the credentials secret isn't set.
+		// This allows authentication via IAM roles.
+		// More info: https://github.com/percona/k8spxc-docs/blob/87f98e6ddae8114474836c0610155d05d3531e03/docs/backups-storage.md?plain=1#L116-L126
+		return nil
+	}
+
+	return s.restorerOptions.ValidateJob(ctx, job)
+}
+
+func (s *s3) Validate(ctx context.Context) error {
+	sec := corev1.Secret{}
+	err := s.k8sClient.Get(ctx,
+		types.NamespacedName{Name: s.bcp.Status.S3.CredentialsSecret, Namespace: s.bcp.Namespace}, &sec)
+	if client.IgnoreNotFound(err) != nil {
+		return errors.Wrap(err, "failed to get secret")
+	}
+
+	accessKeyID := string(sec.Data["AWS_ACCESS_KEY_ID"])
+	secretAccessKey := string(sec.Data["AWS_SECRET_ACCESS_KEY"])
+	ep := s.bcp.Status.S3.EndpointURL
+	bucket, prefix := s.bcp.Status.S3.BucketAndPrefix()
+	verifyTLS := true
+	if s.bcp.Status.VerifyTLS != nil && !*s.bcp.Status.VerifyTLS {
+		verifyTLS = false
+	}
+	if s.cluster.Spec.Backup != nil && len(s.cluster.Spec.Backup.Storages) > 0 {
+		storage, ok := s.cluster.Spec.Backup.Storages[s.bcp.Spec.StorageName]
+		if ok && storage.VerifyTLS != nil {
+			verifyTLS = *storage.VerifyTLS
+		}
+	}
+	s3cli, err := s.newStorageClient(&storage.S3Options{
+		Endpoint:        ep,
+		AccessKeyID:     accessKeyID,
+		SecretAccessKey: secretAccessKey,
+		BucketName:      bucket,
+		Prefix:          prefix,
+		Region:          s.bcp.Status.S3.Region,
+		VerifyTLS:       verifyTLS,
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to create s3 client")
+	}
+	dest := s.bcp.Status.Destination
+	dest = strings.TrimPrefix(dest, api.AwsBlobStoragePrefix)
+	dest = strings.TrimPrefix(dest, bucket+"/")
+	if prefix != "" {
+		dest = strings.TrimPrefix(dest, prefix)
+		dest = strings.TrimPrefix(dest, "/")
+	}
+	dest = strings.TrimSuffix(dest, "/") + "/"
+
+	objs, err := s3cli.ListObjects(ctx, dest)
+	if err != nil {
+		return errors.Wrap(err, "failed to list objects")
+	}
+	if len(objs) == 0 {
+		return errors.New("backup not found")
+	}
+
+	return nil
+}
+
+type pvc struct{ *restorerOptions }
+
+func (s *pvc) Validate(ctx context.Context) error {
+	destination := s.bcp.Status.Destination
+
+	pod, err := backup.PVCRestorePod(s.cr, s.bcp.Status.StorageName, strings.TrimPrefix(destination, "pvc/"), s.cluster)
+	if err != nil {
+		return errors.Wrap(err, "restore pod")
+	}
+	if err := k8s.SetControllerReference(s.cr, pod, s.scheme); err != nil {
+		return err
+	}
+	pod.Name += "-verify"
+	pod.Spec.Containers[0].Command = []string{"bash", "-c", `[[ $(stat -c%s /backup/xtrabackup.stream) -gt 1048576 ]]`}
```

Comment:
after taking backup we do the [same validation ](https://github.com/percona/percona-docker/blob/main/percona-xtradb-cluster-8.0-backup/run_backup.sh#L93)and mark backup as failed if it's smaller than 5000000 bytes

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/1403/commits/95043a1ebaf8085ed9a4567d4a50b71c5fa1d851